NetKet/Sampler/abstract_sampler.hpp

```diff
@@ -50,8 +51,35 @@ class AbstractSampler {
     this->Reset(true);
   }
 
+  virtual void SetMachineFunc(
+      std::function<double(const Complex&)> machine_func) {
+    machine_func_.Set(machine_func);
+  }
+
+  virtual std::function<double(const Complex&)> GetMachineFunc() {
```

Comment:
This should return a const reference, be marked `const noexcept`, and shouldn't be virtual.